modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/viewsReducer.js

```diff
@@ -41,6 +42,21 @@ VIEWS_ACTIONS[VIEWS_ACTION_TYPES.ADD_OR_UPDATE_CUSTOM_VIEW] = (
 	};
 };
 
+VIEWS_ACTIONS[VIEWS_ACTION_TYPES.BATCH_UPDATE] = (state, stateUpdates) => {
+	if (!Array.isArray(stateUpdates) || !stateUpdates.length) {
+		return state;
+	}
+
+	return stateUpdates.reduce((acc, current) => {
+		const {type, value} = current;
+		if (!VIEWS_ACTIONS[type]) {
+			return acc;
+		}
+
+		return VIEWS_ACTIONS[type](acc, value);
+	}, state);
+};
+
```

Comment:
We might not need this, please see [this table](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/3891233068/LPD-55736+-+Research+the+best+way+of+retrieving+and+setting+data+in+the+URL+to+reconcile+with+Data+Set+state#An-SPA-within-an-(senna)-SPA) with the combinations, this only applies when SPA is off and we are pushing states instead of replacing. Still, back button will take user to a state where only 1 change takes place. Instead of comparing that state with the FDS internal state and make just that state change, I'm batching them all here

Reply:
Still, I like how code looks like...